Gems/ROS2/Code/Source/RobotImporter/URDF/UrdfParser.cpp

```diff
@@ -93,13 +94,25 @@ namespace ROS2::UrdfParser
         ParseResult parseResult;
         std::ostringstream parseStringStream;
         {
-            RedirectSDFOutputStream redirectConsoleStream(sdf::Console::Instance()->GetMsgStream(), parseStringStream);
+            RedirectSDFOutputStream redirectConsoleStreamMsg(sdf::Console::Instance()->GetMsgStream(), parseStringStream);
             parseResult.m_sdfErrors = parseResult.m_root.LoadSdfString(xmlString, parserConfig);
             // Get any captured sdf::Error messages
         }
+        // escape console's color codes
+        // Define a regex pattern to match ANSI escape sequences
 
-        std::string parseMessages = parseStringStream.str();
-        parseResult.m_parseMessages = AZStd::string(parseMessages.c_str(), parseMessages.size());
+        // The pattern matches the following ANSI escape sequences:
+
+        // escape console's color codes
```

Comment:
Seems like this comment is here twice and there's redundant whitespace / newlines